RockWeb/Themes/NewSpring/assets/Lava/LAYOUT/card-list.lava

```diff
@@ -1,3 +1,23 @@
+{% comment %}
+    This loops through the items from the dataset and strips out anything that is future-dated or expired.
+{% endcomment %}
+{%- capture displayItemsString -%}
+    {%- assign now = 'Now' | Date:'yyyyMMddHHmmss' -%}
+    [{%- for item in Items -%}
+        {%- assign status = item.Status %}
+        {%- assign publishDateTime = item.PublishDateTime | Date:'yyyyMMddHHmmss' -%}
+        {%- assign expireDateTime = item.ExpireDateTime | Date:'yyyyMMddHHmmss' -%}
+        {%- if status == 2 and publishDateTime <= now and expireDateTime == null or status == 2 and publishDateTime <= now and expireDateTime > now  -%}
+            {{ item | ToJSON }},
+        {%- endif -%}
+    {%- endfor -%}]
+{%- endcapture -%}
+
+{% comment %}
+    This reassigns the results of the above filtered loop to the "Items" object that is used by the card list below.
+{% endcomment %}
+{% assign Items = displayItemsString | ReplaceLast:',', '' | StripNewlines | FromJSON %}
+
```

Comment:
Commented inline, but basically this re-builds the lava object that contains the items' data. For arrays in lava, the only filter is an equality filter which prevents us from being able to filter out based on greater than/less than publish/expire dates. This is the only way I could come up with for us to handle it.